docs/enhancements/installation-progress-bar.md

```diff
@@ -0,0 +1,113 @@
+  * that will also make it usable from other APIs as kube-api
+* Installed clusters with failed non-essential workers or OLM-operators should still reach 100% progress.
+
+
+## Current implementation
+
+* Computed in UI
+* `progressPercent = hostsProgressPercent * 0.75 + operatorsProgressPercent * 0.25`
+  * `operatorsProgressPercent =(completedOperators / monitoredOperators) * 100`
+    * A completedOperator is one of the following:
+      * A built-in operator with `available` status
+      * An OLM operator with `available` or `failed` status
+  * `hostsProgressPercent = (ΣcompletedStages / ΣallStages) * 100`
+    * For each host the data is retrieved by using the following APIs
+
+```
+GET api/assisted-install/v1/clusters/{cluster_id} | jq '.hosts[0].progress'
+{
+  "current_stage": "Done",
+  "stage_started_at": "2021-05-31T13:37:26.224Z",
+  "stage_updated_at": "2021-05-31T13:37:26.224Z"
+}
+
+GET /api/assisted-install/v1/clusters/{cluster_id} | jq '.hosts[0].progress_stages'
+[
+  "Starting installation",
+  "Installing",
+  "Writing image to disk",
+  "Waiting for control plane",
+  "Rebooting",
+  "Configuring",
+  "Joined",
+  "Done"
+]
+```
+
+## New implementation
+
+* Will be computed in BE
+* `progres =ΣclusterStageProgress * Wi = preparingForInstallation * Wpi + installing * Wi + finalizing * Wf` while `ΣWi=1`
+  * We can manually compute an avg duration for the previous installations based on events to choose those weights and hardcode them.
+
+
+#### Preparing for installation
+
+This stage mostly generates manifests and installation configs, therefore, we will set  `preparingForInstallationProgress={1, 0}` depending on if it is done or not.
+
+
+#### Installing
+
+* This stage refers to hosts' installation
+* We will “assume” all host’s stages have a similar duration, therfore, they will be equally weighted.  
+* Hosts' progresses will be computed the same way we it is done in the current implementation, aka `hostsProgressPercent = ΣcompletedStages / ΣallStages`
+
+
+#### Finalizing
+
+* This stage refers to operators' installation
+* We will “assume” all operatos have a similar installation duration.
+* Operators' progresses will be computed the same way we it is done in the current implementation, aka `operatorsProgressPercent = completedOperators / monitoredOperators`
+
+Suggested APIs:
+
+```
+GET /api/assisted-install/v1/clusters/{cluster_id}
+{
+  "id": "3a745598-dde4-46dd-bdb1-ce7e1dea119b",
+  "progress":                                    <-------- NEW
+  {
+    "total_percent": "21",
+    "preparing_for_installation_percent": "100",
+    "installing_percent": "38",
+    "finalizing_percent": "0",
+  }
+  "hosts": [
+    {
+      "kind": "Host",
+      "id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
+      "role": "master",
+      "progress": {
+        "installation_percent": "15",                  <-------- NEW (but coming from hosts API)
+        "current_stage": "Starting installation",
```

Comment:
I don't agree with the "current_stage" field semantics.  
As it is presented right now I can argue it is providing a description/explanation about the stage. If this is intentional, then its place is with the API documentation. 
If this field is meant to actually represent the host's current stage then its value must be constrained to be one of all possible stages a host can be at any moment during the installation. Please consider changing this to be an enum/iota value if this is the case.  
Right now this value is not something the UI can work with:
1. The content might need a microcopy in order to improve the UX
2. The casing of the words in the sentence might not follow UX best practices (this is a presentation layer concern)
3. Establishing the current stage in the FE code demands parsing a potentially large string.

Being that said, I understand that some consumers of the API will find value in having a description provided by the same API for the host's current stage. In this case, I suggest adding an additional field (called "description" maybe?) that can be optionally added to the API response by adding some URL params in the GET request, for example: `GET /api/assisted-install/v1/clusters/{cluster_id}?description=true`.

Reply:
> If this field is meant to actually represent the host's current stage then its value must be constrained to be one of all possible stages a host can be at any moment during the installation

This is the case, please see https://github.com/openshift/assisted-service/blob/ad7260c5a8583fd945fd58096d617cb5ebfec669/swagger.yaml#L4483

Anyway, this is the current API (not part of the design), I just added it for "sanity", the new parts are marked by the `NEW` marker with `<----`.

I don't think we should address that in this design, if you think it should be changed, please consult with @gamli75  and if needed let us have a different task to change it.

Reply:
> I don't think we should address that in this design, if you think it should be changed, please consult with
> @gamli75 and if needed let us have a different task to change it.
I agree with you on this we shouldn't tackle this issue here. I thought it was a new addition :)

Reply:
Great :) 
Resolving the thread.